projects/ngx-charts-on-fhir/src/lib/data-layer/data-layer-color.service.spec.ts

```diff
@@ -0,0 +1,100 @@
+import { TestBed, inject } from '@angular/core/testing';
+import { COLOR_PALETTE, DataLayerColorService } from './data-layer-color.service';
+
+describe('DataLayerColorService', () => {
+  let colorService: DataLayerColorService;
+  let palette: string[] = ['#000000', '#ffffff'];
+
+  beforeEach(async () => {
+    colorService = new DataLayerColorService(palette);
+    await TestBed.configureTestingModule({
+      providers: [
+        { provide: DataLayerColorService, useValue: colorService },
+        { provide: COLOR_PALETTE, useValue: ['#000000', '#ffffff'] },
+      ],
+    }).compileComponents();
+  });
+
+  describe('chooseColorsFromPalette', () => {
+    it('should call chooseColorsFromPalette', inject([DataLayerColorService], (service: DataLayerColorService) => {
+      const layer: any = {
+        datasets: [{ label: 'Diastolic Blood Pressure' }],
+        annotations: [{ label: { display: true } }],
+        backgroundColor: '#ECF0F9',
+      };
+      let setColorSpy = spyOn(service, 'setColor');
+      service.chooseColorsFromPalette(layer);
+      expect(setColorSpy).toHaveBeenCalled();
+    }));
+  });
+  describe('getAnnotationColor', () => {
+    it('should get correct annotation color by calling getAnnotationColor', inject([DataLayerColorService], (service: DataLayerColorService) => {
+      const annotation: any = {
+        label: {},
+        backgroundColor: '#ECF0F9',
+      };
+      expect(service.getAnnotationColor(annotation)).toEqual(annotation.backgroundColor);
+    }));
+  });
+
+  describe('getColor', () => {
+    it('should getColor called', () => {
+      const dataset: any = {
+        label: 'Diastolic Blood Pressure',
+        yAxisID: 'mm[Hg]',
+        data: [],
+        backgroundColor: '#e41a1c33',
+      };
+      const getColorSpy = spyOn(colorService, 'getColor');
+      colorService.getColor(dataset);
+      expect(getColorSpy).toBeDefined();
+      expect(getColorSpy).toHaveBeenCalled();
+    });
```

Comment:
@amolht @yashpmohod
This test is not testing anything. It is calling a stub.
Tests should not spyOn/stub/mock the code that they are testing.
Test should not make direct calls to their own spies.

The purpose of spies/stubs/mocks is to replace the code's _dependencies_ with simplified ones that we can observe and control.

Test should call the _real_ code. The real code will call the spied-on dependencies. Then the test can check that the correct dependencies were called.